docs/source/en/optimization/tgate.md

```diff
@@ -0,0 +1,194 @@
+# TGATE
+
+[TGATE](https://github.com/HaozheLiu-ST/T-GATE/tree/main) accelerates inferences of [`PixArtAlphaPipeline`], [`StableDiffusionPipeline`], and [`StableDiffusionXLPipeline`] by skipping the calculation of cross-attention once it converges. More details can be found at [technical report](https://huggingface.co/papers/2404.02747).
+
+![](https://github.com/HaozheLiu-ST/T-GATE/assets/53887227/bff43e0e-2372-4edc-9ba1-64dcbc649329)
+
+
+
+
+## 🚀 Major Features 
+
+* Training-Free.
+* Easily Integrate into [Diffusers](https://github.com/huggingface/diffusers/tree/main).
+* Only a few lines of code are required.
+* Complementary to [DeepCache](https://github.com/horseee/DeepCache).
+* Friendly support [Stable Diffusion pipelines](https://huggingface.co/stabilityai), [PixArt](https://pixart-alpha.github.io/), and [Latent Consistency Models](https://latent-consistency-models.github.io/).
+* 10%-50% speed up for different models. 
+
+## 📖 Quick Start
+
+### 🛠️ Installation
+
+Start by installing [TGATE](https://github.com/HaozheLiu-ST/T-GATE/tree/release-v.0.1.0):
+
+```
+pip install tgate
+```
+
+#### Requirements
+
+* pytorch>=2.0.0
+* diffusers>=0.27.2
+* transformers==4.37.2
+* DeepCache==0.1.1
+* accelerate
+
+### 🌟 Usage
+
+Accelerate `PixArtAlphaPipeline` with TGATE:
+
+```diff
+import torch
+from diffusers import PixArtAlphaPipeline
+
+pipe = PixArtAlphaPipeline.from_pretrained("PixArt-alpha/PixArt-XL-2-1024-MS", torch_dtype=torch.float16)
+
++ from tgate import TgatePixArtLoader
++ gate_step = 8
++ inference_step = 25
++ pipe = TgatePixArtLoader(
++        pipe,
++        gate_step=gate_step,
++        num_inference_steps=inference_step,
++ )
+pipe = pipe.to("cuda")
+
++ image = pipe.tgate(
++         "An alpaca made of colorful building blocks, cyberpunk.",
++         gate_step=gate_step,
++         num_inference_steps=inference_step,
++ ).images[0]
+```
```

Comment:
1. It'd be easier for users to copy/paste code snippet without `diff`  
2. We can also wrap all these pipeline examples with our doc-builders special syntax so users can easily switch between the different code samples without scrolling.

```md
<hfoptions id="pipelines">
<hfoption id="PixArt">

code here

</hfoption>
<hfoption id="Stable Diffusion XL">

code here

</hfoption>
<hfoption id="StableDiffusionXL with DeepCache">

code here

</hfoption>
<hfoption id="Latent Consistency Model">

code here

</hfoption>
</hfoptions>
```

```suggestion
```py
import torch
from diffusers import PixArtAlphaPipeline
from tgate import TgatePixArtLoader

pipe = PixArtAlphaPipeline.from_pretrained("PixArt-alpha/PixArt-XL-2-1024-MS", torch_dtype=torch.float16)
pipe = TgatePixArtLoader(
       pipe,
       gate_step=8,
       num_inference_steps=25,
)
pipe = pipe.to("cuda")

image = pipe.tgate(
       "An alpaca made of colorful building blocks, cyberpunk.",
        gate_step=gate_step,
       num_inference_steps=inference_step,
).images[0]
```
```